src/app/CommandHandler.cpp

```diff
@@ -467,7 +467,17 @@ void CommandHandler::AddStatus(const ConcreteCommandPath & aCommandPath, const P
 {
     // Return prematurely in case of requests targeted to a group that should not add the status for response purposes.
     VerifyOrReturn(!IsGroupRequest());
-    VerifyOrDie(FallibleAddStatus(aCommandPath, aStatus, context) == CHIP_NO_ERROR);
+
+    CHIP_ERROR error = FallibleAddStatus(aCommandPath, aStatus, context);
+
+    if (error != CHIP_NO_ERROR)
+    {
+        ChipLogError(DataManagement, "Failed to add command status: %" CHIP_ERROR_FORMAT, error.Format());
+
+        // Do not crash if the status has not been added due to running out of packet buffers or other resources.
+        // It is better to drop a single response than to go offline and lose all sessions and subscriptions.
+        VerifyOrDie(error == CHIP_ERROR_NO_MEMORY);
```

Comment:
So we crash only if the error is other than CHIP_ERROR_NO_MEMORY? Judging by the comment I expected to cash exactly in such a case. Or am I lost here? :)

Reply:
I think the comment is correct: Do **not** crash in the case of failure due to "running out of...". But if you have a proposal how to rephrase that to make it clearer for you, spit it out :).

Reply:
discussed f2f.